Source/Meadow.Foundation.Peripherals/Sensors.Environmental.NextPm/Driver/NextPm.cs

```diff
@@ -0,0 +1,295 @@
+﻿using Meadow.Hardware;
+using Meadow.Units;
+using System;
+using System.Threading.Tasks;
+
+namespace Meadow.Foundation.Sensors.Environmental
+{
+    /// <summary>
+    /// Represents a TERA Sensor NextPM particulate matter sensor
+    /// </summary>
+    public partial class NextPm :
+        PollingSensorBase<(
+            ParticulateReading? reading10s,
+            ParticulateReading? reading1min,
+            ParticulateReading? reading15min,
+            Units.Temperature? temperature,
+            RelativeHumidity? humidity)>,
+        IDisposable, IPowerControllablePeripheral
+    {
+        /// <summary>
+        /// Raised when a new 10-second average reading is taken
+        /// </summary>
+        public event EventHandler<IChangeResult<ParticulateReading>> Readings10sUpdated = delegate { };
+        /// <summary>
+        /// Raised when a new 1-minute average reading is taken
+        /// </summary>
+        public event EventHandler<IChangeResult<ParticulateReading>> Readings1minUpdated = delegate { };
+        /// <summary>
+        /// Raised when a new 15-minute average reading is taken
+        /// </summary>
+        public event EventHandler<IChangeResult<ParticulateReading>> Readings15minUpdated = delegate { };
+        /// <summary>
+        /// Raised when a new temperature reading is taken
+        /// </summary>
+        public event EventHandler<IChangeResult<Units.Temperature>> TemperatureUpdated = delegate { };
+        /// <summary>
+        /// Raised when a new humidity reading is taken
+        /// </summary>
+        public event EventHandler<IChangeResult<RelativeHumidity>> HumidityUpdated = delegate { };
+
+        /// <summary>
+        /// Returns true if the object is disposed, otherwise false
+        /// </summary>
+        public bool IsDisposed { get; private set; }
+
+        /// <summary>
+        /// Creates a NextPm instance
+        /// </summary>
+        /// <param name="portName"></param>
+        public NextPm(SerialPortName portName)
+            : this(portName.CreateSerialPort())
+        {
+        }
+
+        /// <summary>
+        /// Creates a NextPm instance
+        /// </summary>
+        /// <param name="port"></param>
+        public NextPm(ISerialPort port)
+        {
+            _port = InitializePort(port);
+
+            _port.Open();
+        }
+
+        /// <summary>
+        /// Gets the sensor's firmware
+        /// </summary>
+        /// <returns></returns>
```

Comment:
Can we fill out what does it return for each function in this class? Otherwise, remove ///<returns></returns> to avoid having empty XML tag values.